src/atama/sender/node/sender_node.cpp

```diff
@@ -0,0 +1,178 @@
+// SOFTWARE.
+
+#include <bits/stdc++.h>
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "atama/sender/node/sender_node.hpp"
+#include "rclcpp/rclcpp.hpp"
+#include "tachimawari_interfaces/msg/set_joints.hpp"
+
+using namespace std::chrono_literals;
+
+namespace atama::sender
+{
+
+SenderNode::SenderNode(rclcpp::Node::SharedPtr node, std::shared_ptr<atama::head::Head> head)
+: node(node), head(head)
+{
+  set_joints_publisher = node->create_publisher<tachimawari_interfaces::msg::SetJoints>(
+    "/joint/set_joints", 10);
+  set_head_publisher = node->create_publisher<atama_interfaces::msg::Head>(
+    "/head_data/set_head_data", 10);
+}
+
+void SenderNode::publish_joints()
+{
+  auto joints_msg = tachimawari_interfaces::msg::SetJoints();
+
+  const auto & joints = head->get_joints();
+  auto & joint_msgs = joints_msg.joints;
+
+  joint_msgs.resize(joints.size());
+  for (size_t i = 0; i < joints.size() && i < joint_msgs.size(); ++i) {
+    joint_msgs[i].id = joints[i].get_id();
+    joint_msgs[i].position = joints[i].get_position();
+  }
+
+  set_joints_publisher->publish(joints_msg);
+}
+
+void SenderNode::publish_head_data()
+{
+  auto pan_tilt_msg = atama_interfaces::msg::Head();
+
+  pan_tilt_msg.pan_angle = head->get_pan_angle();
+  pan_tilt_msg.tilt_angle = head->get_tilt_angle();
+  pan_tilt_msg.distance = head->calculate_distance_from_pan_tilt(
+    head->get_pan_angle(), head->get_tilt_angle());
+
+  set_head_publisher->publish(pan_tilt_msg);
+}
+
+void SenderNode::process(int function_id)
+{
+  using atama::head::Head;
+
+  if (!head->is_joint_empty()) {
+    switch (function_id) {
+      case Head::SCAN_UP: head->scan_up(); break;
+      case Head::SCAN_DOWN: head->scan_down(); break;
+      case Head::SCAN_VERTICAL: head->scan_vertical(); break;
+      case Head::SCAN_HORIZONTAL: head->scan_horizontal(); break;
+      case Head::SCAN_MARATHON: head->scan_marathon(); break;
+      case Head::SCAN_CUSTOM: head->scan_custom(Head::SCAN_CUSTOM); break;
+      case Head::TRACK_OBJECT:
+        head->track_object(head->get_object_name());
+        break;
+      case Head::MOVE_BY_ANGLE:
+        head->move_by_angle(
+          head->get_pan_angle_goal(),
+          head->get_tilt_angle_goal());
+        break;
+        // TODO(nathan): implement look_to_position() after
+        // robot_position_x, robot_position_y, yaw are obtained
+        // case Head::LOOK_TO_POSITION:
+        //   head->look_to_position();
+        //   break;
+    }
+  }
+
+  publish_joints();
+  publish_head_data();
+}
+
+bool SenderNode::check_scan()
+{
+  std::vector<std::string> result_name;
+  for (const auto & name : head->detection_result) {
+    result_name.push_back(name.label);
+  }
+
+  std::vector<std::string>::iterator it;
+  it = std::find(result_name.begin(), result_name.end(), head->get_object_name());
+
+  // Object Found
+  if (it != result_name.end()) {
+    return true;
+  }
+  return false;
```

Comment:
if all what you done is just finding whether label of `head->detection_result` contains `head->get_object_name()`, i suggest to use `std::set` instead of `std::vector`. later you could just use `set.find()` instead of `std::find()`.

see this line too
https://github.com/ichiro-its/atama/blob/fdd5eeceff0fc9eb8131e6839c22fd8620d3f4ca/src/atama/sender/node/sender_node.cpp#L124-L138

Reply:
done